vars/stageDeployToOpenshift.txt

```diff
@@ -0,0 +1,19 @@
+stageDeployToOpenshift deploys the service.
+
+A deployment is more than just updating an image, it might also involve adding
+an environment variable, or changing constraints etc. Because of this, a
+deployment should be one atomic JSON patchset.
+
+When an "openshift" folder is present, we use the "tailor" binary in the
+slave-base to apply its templates. The behaviour is customizable by
+adding a "Tailorfile" (or a more specific, per-project version of it).
+
```

Comment:
On the other hand tailor computes JSON patchset comparing the local definition with the current infrastructure.   So I got a bit confused what was meant with the initial JSON patchset mentioned.

Furthermore I assumed that one creates a JSON patchset so that openshift sees all changes at once (atomic). However the next sentence mentions the triggers which confused me. 

I believe what is envisioned however is that the openshift folder contains the entire openshift resources for a project and one can now make a change by either applying a new TAGVERSION or making some config change in unrelated areas such as a secret or configuration maps which are referenced by a deployment. 
Is this what is envisioned here?

Reply:
The `openshift` folder would only contain templates (desired state) of the component, not the whole project. This is a potential downside of the approach taken here.

A "change" is simply a Git commit - and this might be a change in some e.g. Java code that then gets deployed via the image, or it might be a change in the OCP template, which then gets deployed via the JSON patchsets that Tailor generates from it.

The vision is that all changes to the component are done in one repo, and deployed in one Jenkins pipeline - no manual work via different means.